cmd/kubeadm/app/phases/etcd/local.go

```diff
@@ -203,7 +203,7 @@ func GetEtcdPodSpec(cfg *kubeadmapi.ClusterConfiguration, endpoint *kubeadmapi.A
 					v1.ResourceMemory: resource.MustParse("100Mi"),
 				},
 			},
-			LivenessProbe: staticpodutil.LivenessProbe(probeHostname, "/health", probePort, probeScheme),
+			LivenessProbe: staticpodutil.LivenessProbe(probeHostname, "/health?exclude=NORAFTLEADER&consistency=s", probePort, probeScheme),
```

Comment:
If I got the problem right, we are trying to solve problems happening when restarting a cluster.
Given this context, I'm not sure how/why we manage to pass startupProbe check that uses /health, while instead we have to relax /health on liveness.

Also, what are the drawbacks of relaxing this check on a running cluster (not being restarted)?

Reply:
As @neolit123 pointed out that `I was under the impression that it does not work.`, I need to do some testings around this.

Is there a simple way to simulate such a scenario？

Reply:
@randomvariable knows more details.

I think we should wait for the etcd maintainers to reapond before merging changes in kubeadm. The etcd issue is linked on the kubeadm issue.

Reply:
We haven't actually made any changes to etcd yet to wire this up, I was hoping an etcd maintainer would bless the approach before ploughing through, though the change looks easy enough to do in etcd itself.